src/lib/y2storage/autoinst_profile/drive_section.rb

```diff
@@ -204,7 +204,10 @@ def init_from_disk(disk)
         return false if disk.partitions.empty?
 
         @type = :CT_DISK
+        # FIXME: could anyone with knowledge leave a comment why s390 is special here?
         @device = Yast::Arch.s390 ? disk.udev_full_paths.first : disk.name
+        # if disk.udev_full_paths.first is nil go for disk.name anyway
+        @device ||= disk.name
```

Comment:
as it goes this complex maybe group it with:
```
@device = 
  if Yast::Arch.s390
    disk.udev_full_paths.first || disk.name
  else
    disk.name
  end
```

so fallback is used only in s390 case.

Reply:
I've tried a few variants already and I had to go with one rubocop likes.